app/screens/edit_profile/edit_profile.test.tsx

```diff
@@ -76,25 +88,26 @@ describe('EditProfile', () => {
             />,
         );
 
-        await act(async () => {
-            await new Promise((resolve) => setTimeout(resolve, 0));
+        await waitFor(() => {
+            const {fetchCustomAttributes} = require('@actions/remote/user');
+            expect(fetchCustomAttributes).toHaveBeenCalledWith('http://localhost:8065', 'user1');
         });
 
-        const {fetchCustomAttributes} = require('@actions/remote/user');
-        expect(fetchCustomAttributes).toHaveBeenCalledWith('http://localhost:8065', 'user1');
-        const TEST_ID = 'edit_profile_form.customAttributes.attr1.input';
+        const customAttributeItems = await findAllByTestId(new RegExp('^edit_profile_form.customAttributes.attr[0-9]+.input$'));
+        expect(customAttributeItems.length).toBe(3);
 
-        const input = getByTestId(TEST_ID);
+        expect(customAttributeItems[0].props.value).toBe('original value 1');
+        expect(customAttributeItems[1].props.value).toBe('original value 2');
+        expect(customAttributeItems[2].props.value).toBe('original value 3');
 
         await act(async () => {
-            fireEvent.changeText(input, 'new value');
+            fireEvent.changeText(customAttributeItems[1], 'new value');
         });
 
-        const updatedCustomAttr = getByTestId(TEST_ID);
-
-        expect(updatedCustomAttr.props.value).toBe('new value');
-
-        expect(updatedCustomAttr.props.name).toBe(input.props.name);
-        expect(updatedCustomAttr.props.sort_order).toBe(input.props.sort_order);
+        const newCustomAttributeItems = await findAllByTestId(new RegExp('^edit_profile_form.customAttributes.attr[0-9]+.input$'));
+        expect(newCustomAttributeItems.length).toBe(3);
+        expect(newCustomAttributeItems[0].props.value).toBe('original value 1');
+        expect(newCustomAttributeItems[1].props.value).toBe('new value');
+        expect(newCustomAttributeItems[2].props.value).toBe('original value 3');
```

Comment:
I like this better.  It gives me the sense that the order is still intact, and seeing the 'new value' as well.